### PR TITLE
Int2019 not displaying view profile in admin view

### PR DESCRIPTION
**Pull request by moleguaco99**

### CODE REVIEW

File: ClientApp/viking-vault/src/Common/AdminIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import icon from "../Resources/images/user.png";
+import "./styles.css";
+import {Redirect} from "react-router-dom";
+import { constants } from "../Resources/Constants";
+import UserIconAvatar from "./UserIconAvatar"
+
+const API_URL = `${constants.baseUrl}userprofilepages`;
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/Common/AdminIcon.tsx]: rename to a more meaningful name

File: ClientApp/viking-vault/src/Common/AdminIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    document.removeEventListener("click", this.handleOutsideClick, false);
+  }
+
+  handleClick = () =>{
+    this.setState({ clicked: !this.state.clicked });
+  }
+
+  handleOutsideClick = (event: Event) => {
+    if(this.megaMenu == null || this.megaMenu.current == null)
+      return;
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/Common/AdminIcon.tsx]: move return on the same line with if, to be more readable

File: ClientApp/viking-vault/src/Common/AdminIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      adminImage: icon
+    };
+    this.megaMenu = React.createRef();
+    this.handleOutsideClick = this.handleOutsideClick.bind(this);
+  }
+
+  sendData = () =>{
+    setTimeout(() => {
+      this.props.parentCallBack(false);
+    }, 100);
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/Common/AdminIcon.tsx:33]: What does this `100` represents?

File: ClientApp/viking-vault/src/Common/AdminIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  componentDidMount() {
+    document.addEventListener("click", this.handleOutsideClick, false);
+
+    let token = sessionStorage.getItem("Authentication-Token");
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/Common/AdminIcon.tsx]: declare it with const, is not changed, therefore is readonly variable

File: ClientApp/viking-vault/src/Common/AdminIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+  sendData = () =>{
+    setTimeout(() => {
+      this.props.parentCallBack(false);
+    }, 100);
+  }
+
+  componentDidMount() {
+    document.addEventListener("click", this.handleOutsideClick, false);
+
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/Common/AdminIcon.tsx]: remove empty line

**Review by crisfsabau — CHANGES_REQUESTED** [review]

File: ClientApp/viking-vault/src/Common/AdminIcon.tsx (hunk 3 above)
Comment (moleguaco99) [reply ClientApp/viking-vault/src/Common/AdminIcon.tsx:33]: It is a timeout that allows for a state changing of a referred component before rendering another component

**Pull request merged**